views/devices/form.erb

```diff
@@ -1,36 +1,9 @@
-    </div>
-
-    <% if device.persisted? %>
-      <div class="p-6">
-        <label for="device_refresh_interval" class="block text-gray-700 text-md leading-4 font-medium mb-2">Refresh Interval</label>
-        <input type="text" required="true" name="device[refresh_interval]" id="device_refresh_interval" value="<%= device.refresh_interval %>" class="block mt-2 min-h-12 px-4 rounded-xl text-black bg-white block w-full p-0 outline-none text-sm border border-gray-300 focus:outline-none ring-2 ring-transparent focus:ring-2 ring-offset-gray-200 focus:ring-offset-2 focus:ring-offset-gray-200 focus:ring-blue-500 focus:border-transparent transition duration-150" placeholder="qwerty_asdf_420">
-      </div>
-      <div class="p-6">
-        <label for="device_friendly_id" class="block text-gray-700 text-md leading-4 font-medium mb-2">Friendly ID</label>
-        <input type="text" required="true" name="device[friendly_id]" id="device_friendly_id" value="<%= device.friendly_id %>" class="block mt-2 min-h-12 px-4 rounded-xl text-black bg-white block w-full p-0 outline-none text-sm border border-gray-300 focus:outline-none ring-2 ring-transparent focus:ring-2 ring-offset-gray-200 focus:ring-offset-2 focus:ring-offset-gray-200 focus:ring-blue-500 focus:border-transparent transition duration-150" placeholder="qwerty_asdf_420">
-      </div>
-      <div class="p-6">
-        <label for="device_api_key" class="block text-gray-700 text-md leading-4 font-medium mb-2">API Key</label>
-        <input type="text" required="true" name="device[api_key]" id="device_api_key" value="<%= device.api_key %>" class="block mt-2 min-h-12 px-4 rounded-xl text-black bg-white block w-full p-0 outline-none text-sm border border-gray-300 focus:outline-none ring-2 ring-transparent focus:ring-2 ring-offset-gray-200 focus:ring-offset-2 focus:ring-offset-gray-200 focus:ring-blue-500 focus:border-transparent transition duration-150" placeholder="qwerty_asdf_420">
-      </div>
-    <% end %>
-
-    <div class="p-6 text-right">
-      <input type="submit" name="commit" value="Save" class="cursor-pointer font-medium rounded-lg text-sm px-3 py-2 inline-flex items-center transition duration-150 justify-center shrink-0 gap-1.5 whitespace-nowrap text-white bg-blue-500 hover:bg-blue-600 focus:outline-none">
-    </div>
-  </div>
-</form>
\ No newline at end of file
+<%= f.input :name, label: "Name", placeholder: "My TRMNL" %>
+<%= f.input :mac_address, label: "MAC Address", placeholder: "xx:xx:xx:xx:xx:xx" %>
+<% if device.persisted? %>
+  <%= f.input :refresh_interval, label: "Refresh Interval", placeholder: "15" %>
+  <%= f.input :friendly_id, label: "Friendly ID", placeholder: "office_desk" %>
+  <%= f.input :api_key, label: "API Key", placeholder: "xxxxxxxxxx" %>
+<% end %>
+<%= f.button 'Save' %>
```

Comment:
The modifications to the form work, but this implies you could intersperse other things into this form and output them, which does not seem to be the case.

This needs a bit of a rethink.

Reply:
Ok - this has been reworked to explicitly use forme, in order to still allow the ability to put markup in the erb form templates and have it show up.